From 32249a22fc39319651e7c23442d37ec837f05764 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Thu, 8 Sep 2022 02:36:33 +0530
Subject: [PATCH] meson: Fix detection of a system-provided proxy-libintl

proxy-libintl defines ngettext() as a define in the header that points
to the actual symbol in the library which is g_libintl_ngettext().
Same with bind_textdomain_codeset().
---
 meson.build | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -2135,7 +2135,6 @@
 # FIXME: glib-gettext.m4 has much more checks to detect broken/uncompatible
 # implementations. This could be extended if issues are found in some platforms.
 libintl_deps = []
-libintl_prefix = '#include <libintl.h>'
 libintl = dependency('intl', required: false)
 if libintl.found() and libintl.type_name() != 'internal'
   # libintl supports different threading APIs, which may not
@@ -2147,7 +2146,7 @@
   #
   # Meson's builtin dependency lookup as of 0.60.0 doesn't check for
   # pthread, so we do this manually here.
-  if cc.has_function('ngettext', dependencies : libintl, prefix: libintl_prefix)
+  if cc.has_function('ngettext', dependencies : libintl)
     libintl_deps += [libintl]
   else
     libintl_iconv = cc.find_library('iconv', required : false)
@@ -2155,7 +2154,7 @@
       libintl_deps += [libintl, libintl_iconv]
     else
       libintl_pthread = cc.find_library('pthread', required : false)
-      if libintl_pthread.found() and cc.has_function('ngettext', dependencies : [libintl, libintl_pthread], prefix: libintl_prefix)
+      if libintl_pthread.found() and cc.has_function('ngettext', dependencies : [libintl, libintl_pthread])
         libintl_deps += [libintl, libintl_pthread]
       else
         libintl = disabler()
@@ -2165,7 +2164,7 @@
 endif
 
 if libintl.found() and libintl.type_name() != 'internal'
-  have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset', dependencies: libintl_deps, prefix: libintl_prefix)
+  have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset', dependencies: libintl_deps)
 else
   # using proxy-libintl fallback
   libintl = dependency('intl', allow_fallback: true)
